src/main/java/org/jfrog/bamboo/deployment/ArtifactoryDeploymentTask.java

```diff
@@ -73,87 +52,56 @@ public TaskResult execute(@NotNull DeploymentTaskContext deploymentTaskContext)
+    }
 
+    private TaskResult doUpload(@NotNull DeploymentTaskContext deploymentTaskContext, BuildLogger buildLogger,
+                                Log bambooBuildInfoLog, ServerConfig serverConfig, String username, String password) {
+        ArtifactoryBuildInfoClient client = new ArtifactoryBuildInfoClient(serverConfig.getUrl(), username, password, bambooBuildInfoLog);
+        String artifactsRootDirectory = deploymentTaskContext.getRootDirectory().getAbsolutePath();
         try {
-            RuntimeTaskDefinition artifactDownloadTask = TaskUtils.findDownloadArtifactsTask(deploymentTaskContext.getCommonContext().getRuntimeTaskDefinitions());
-            FilesCollector filesCollector = new FilesCollector(artifactsRootDirectory, artifactDownloadTask);
-            Map<String, Set<File>> artifacts = filesCollector.getCollectedFiles();
-            Set<DeployDetails> deployDetailsSet = createDeploymentDetailsForArtifacts(artifacts);
-            deploy(deployDetailsSet);
-            result = TaskResultBuilder.newBuilder(deploymentTaskContext).success().build();
+            String spec = getFileSpec(deploymentTaskContext, buildLogger);
+            if (StringUtils.isBlank(spec)) {
+                String err = "Spec file can't be empty";
+                buildLogger.addErrorLogEntry(err);
+                log.error(err);
+                return TaskResultBuilder.newBuilder(deploymentTaskContext).failedWithError().build();
+            }
+            buildLogger.addBuildLogEntry(spec);
+            SpecsHelper specsHelper = new SpecsHelper(bambooBuildInfoLog);
+            specsHelper.uploadArtifactsBySpec(spec, new File(artifactsRootDirectory), new HashMap<>(), client);
+            return TaskResultBuilder.newBuilder(deploymentTaskContext).success().build();
         } catch (Exception e) {
             buildLogger.addErrorLogEntry("Error while deploying artifacts to Artifactory: " + e.getMessage());
-            result = TaskResultBuilder.newBuilder(deploymentTaskContext).failedWithError().build();
+            return TaskResultBuilder.newBuilder(deploymentTaskContext).failedWithError().build();
         } finally {
             client.close();
         }
-        return result;
     }
 
-    /**
-     * Deploy all collected artifacts to Artifactory
-     *
-     * @param deployDetailsSet details for the artifacts we want to deploy
-     * @throws IOException
-     */
-    private void deploy(Set<DeployDetails> deployDetailsSet) throws IOException {
-        for (DeployDetails deployDetails : deployDetailsSet) {
-            buildLogger.addBuildLogEntry("Deploying: " + deployDetails.getArtifactPath() + " to: " + deployDetails.getTargetRepository());
-            client.deployArtifact(deployDetails);
+    private String getFileSpec(@NotNull DeploymentTaskContext deploymentTaskContext, BuildLogger buildLogger) throws IOException {
```

Comment:
We may be able to reuse this code for the 4 tasks which currently use File Specs. Basically in all tasks, we're checking whether the Spec is read from the Task Configuration or a file. 
Having this code in one place can also help in standardise the logging. For example, we'll have a log print with the expected Spec file location in all tasks.